grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -0,0 +1,1655 @@
+                    onMouseOver={(): void => this.onEntityMouseOver(entityName)}
+                    onMouseLeave={(): void => this.onEntityMouseLeave()}
+                    onClick={(): void => this.onEntityClick(entityName)}
+                >
+                    {bars.length ? (
+                        bars.map((bar) => (
+                            <MarimekkoBar
+                                key={`${entityName}-${bar.seriesName}`}
+                                bar={bar}
+                                tooltipProps={{
+                                    ...tooltipProps,
+                                    highlightedSeriesName: bar.seriesName,
+                                }}
+                                barWidth={barWidth}
+                                isHovered={isHovered}
+                                isSelected={isSelected}
+                                isFaint={isFaint}
+                                entityColor={entityColor?.color}
+                                y0={y0}
+                                isInteractive={
+                                    !this.manager.isExportingtoSvgOrPng
+                                }
+                                dualAxis={dualAxis}
+                            />
+                        ))
+                    ) : (
+                        <MarimekkoBar
+                            key={`${entityName}-placeholder`}
+                            tooltipProps={tooltipProps}
+                            bar={{
+                                kind: BarShape.BarPlaceholder,
+                                seriesName: entityName,
+                                height: noDataHeight,
+                            }}
+                            barWidth={barWidth}
+                            isHovered={isHovered}
+                            isSelected={isSelected}
+                            isFaint={isFaint}
+                            entityColor={entityColor?.color}
+                            y0={y0}
+                            isInteractive={!this.manager.isExportingtoSvgOrPng}
+                            dualAxis={dualAxis}
+                        />
+                    )}
+                </g>
+            )
+            if (isSelected || isHovered) highlightedElements.push(result)
+            else normalElements.push(result)
+        }
+
+        return ([] as JSX.Element[]).concat(
+            patterns,
+            noDataAreaElement ? [noDataAreaElement] : [],
+            normalElements,
+            placedLabels,
+            labelLines,
+            highlightedElements,
+            noDataLabel ? [noDataLabel] : []
+        )
+    }
+    private paddingInPixels = 5
+
+    private static labelCandidateFromItem(
+        item: EntityWithSize,
+        baseFontSize: number,
+        isSelected: boolean
+    ): LabelCandidate {
+        return {
+            item: item,
+            bounds: Bounds.forText(item.entityId, {
```

Comment:
nit: this doesn't seem to be `entityId` which tends to be a numeric ID, but it's an `entityName`?